lib/AST/ASTContext.cpp

```diff
@@ -1921,6 +1921,11 @@ ASTContext::getModule(ImportPath::Module ModulePath) {
   auto moduleID = ModulePath[0];
   for (auto &importer : getImpl().ModuleLoaders) {
     if (ModuleDecl *M = importer->loadModule(moduleID.Loc, ModulePath)) {
+      if (LangOpts.EnableModuleLoadingRemarks) {
+        Diags.diagnose(nullptr,
```

Comment:
Is this `nullptr` a null source location? If so, do we have a source location for the import that we can/should print here?

Reply:
Also, I noticed a bunch of calls with `->loadModule` in other places; I think those need to be updated as well. Or we can centralize the logic: say renaming `loadModule` -> `loadModuleInternal`, add a new method `loadModule` which calls `loadModuleInternal`, does this check and emits the path.

Reply:
There should now be a diagnostic location for imports in the sources. The other sources of imports, like dependencies for loaded swiftmodule, could have a remark in the style of `loading "Combine" a dependency of "SwiftUI"`.

I'll look into what can be done for the other `loadModule`. Most of them are for underlying implementations, I've only noted about three different call sites to the top `loadModule`. Each could use a specific remark too.